spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AbstractAmqpOutboundEndpoint.java

```diff
@@ -604,6 +607,28 @@ public Object getUserData() {
 			return this.message;
 		}
 
+		@Override
+		public SettableListenableFuture<Confirm> getFuture() {
+			if (this.userData instanceof CorrelationData) {
+				return ((CorrelationData) this.userData).getFuture();
+			}
+			else {
+				return super.getFuture();
+			}
+		}
+
+		@Override
+		public org.springframework.amqp.core.Message getReturnedMessage() {
+			return super.getReturnedMessage();
```

Comment:
Looks like you are missing here something: since you delegate to the `userData` in the `setReturnedMessage()`, I guess you need to do the same here as well.
Otherwise this override just doesn't make sense.

Reply:
No, in the setter I call super too; but you are correct, this should be removed.